smaps/src/main/java/com/google/smaps/MemoryMap.java

```diff
@@ -58,11 +58,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     // Go through the regions and add range/permissions pairs to the list.
     for (int i = 0; i < regions.size(); i++) {
       Region curR = regions.get(i);
-      String range = curR.startLoc() + " - " + curR.endLoc();
+      String start = formatAddress(curR.startLoc());
+      String end = formatAddress(curR.endLoc());
+      String range = start + " - " + end;
       Object permissions = (Object) curR.permissions();
       Object[] pair = {range, permissions};
       dataArray.add(pair);
     }
     return dataArray;
   }
+
+  /**
+   * Formats the address string to be 16 characters long by adding zeroes to the beginning of the
+   * address to fill it.
+   */
+  static String formatAddress(String address) {
```

Comment:
I think Java has this baked in, probably in String.format()...try String.format("%02X", value)?
Then you probably won't need a helper.